k2/csrc/fsa.h

```diff
@@ -79,9 +79,18 @@ struct ArcHash {
 
   The start-state is always numbered zero and the final-state is always the
   last-numbered state.  However, we represent the empty FSA (the one that
-  accepts no strings) by having no states at all, so `arcs` would be empty.
+  accepts no strings) by having no states at all, so `arcs_indexes` would be
+  empty.
+
+  TODO(haowen): add below comments in the final version:
+    (In Array2 representation) We represent an empty FSA(the one that accepts no
+  strings) by having no states at all, so `size1` would be 0 (As an empty FSA is
+  an initialized Array2 object, so `indexes` would be allocated and has at least
+  one element, but we don't care about it here).
  */
-struct Fsa {
+// TODO(haowen): finally we will remove `arc_indexes` and `arcs`, but for now,
+// we would keep them to replace Fsa with Array2 incrementally.
+struct Fsa : public Array2<Arc *, int32_t> {
```

Comment:
Does this mean that the stride of the underlying memory is always 1 ?

Reply:
yes, for `FSA`.